auth/api/auth/v2/s2s_vptoken.go

```diff
@@ -0,0 +1,44 @@
+package v2
+
+import (
+	"errors"
+	"github.com/labstack/echo/v4"
+	"github.com/nuts-foundation/nuts-node/core"
+	"net/http"
+)
+
+// serviceToService adds support for service-to-service OAuth2 flows,
+// which uses a custom vp_token grant to authenticate calls to the token endpoint.
+// Clients first call the presentation definition endpoint to get a presentation definition for the desired scope,
+// then create a presentation submission given the definition which is posted to the token endpoint as vp_token.
+// The AS then returns an access token with the requested scope.
+// Requires:
+// - GET /presentation_definition?scope=... (returns a presentation definition)
+// - POST /token (with vp_token grant)
+type serviceToService struct {
+}
+
+func (s serviceToService) Routes(router core.EchoRouter) {
+	router.Add("GET", "/public/oauth2/:did/presentation_definition", func(echoCtx echo.Context) error {
```

Comment:
the manually registered paths need to be updated?

Reply:
This needs to move to api.go#Routes anyways